### PR TITLE
[Header] Remove whitespace from JSON

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
Per suggestion of @Mottie, this PR removes the whitespace from the data-serialization of the header/footer data. It also adds a plugin to escape the JSON string so that the document can parse it via `JSON.parse`, which carries a somewhat-significant performance boost.

Per @jbalboni , I was following this article - https://joreteg.com/blog/improving-redux-state-transfer-performance#option-3-turn-your-state-into-js-string-containing-escaped-json. I might try adding the `window.__REDUX__` part for bootstrapping the Redux store in a follow-up PR, but it was leading to too many changes in my work just now. The MegaMenu's data-flow is surprisingly complex.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/18864

## Testing done
- Confirmed the menu loads locally
- Confirmed boost of page size
- Confirmed `generated/headerFooter.json` (downloaded by TeamSite pages) is being generated w/KB savings

## Screenshots
Using master, I ran `yarn build:content` then checked the homepage's file size. Then I switched to this PR's branch and did the same thing. It appears we save about 24 KB. 

### Before changes
image

### After changes
image

#### Escaped JSON
image

## Acceptance criteria
- [ ] Pages are lighter weight

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by Mottie — APPROVED** [review]
Awesome work!

ghe-emoji

**Review by kelsonic — APPROVED** [review]
Awesomeeee 🔥

File: src/site/stages/build/plugins/create-header-footer.js [hunk 1]
```diff
@@ -51,17 +53,22 @@ function createHeaderFooterData(buildOptions) {
...
 
     const drupalMenu = DRUPALS.PREFIXED_ENVIRONMENTS.has(buildOptions.buildtype)
       ? replaceWithDrupalLinks(headerFooter, files)
       : headerFooter;
-    const drupalMenuSerialized = JSON.stringify(drupalMenu, null, 4);
+    const drupalMenuSerialized = JSON.stringify(drupalMenu);
+
+    const drupalMenuSerializedEscaped = jsesc(drupalMenuSerialized, {
+      json: true,
+      isScriptContext: true,
```
Comment (billfienberg) [inline src/site/stages/build/plugins/create-header-footer.js:65]: Leaving this link here as a breadcrumb:
https://github.com/mathiasbynens/jsesc#isscriptcontext

File: src/site/stages/build/plugins/create-header-footer.js [hunk 2]
```diff
@@ -51,31 +19,19 @@ function createHeaderFooterData(buildOptions) {
...
+    const stringified = JSON.stringify(headerFooter);
 
-    Object.keys(files).forEach(file => {
-      if (files[file].isDrupalPage) {
-        // eslint-disable-next-line no-param-reassign
-        files[file].headerFooterData = drupalMenuSerialized;
-      } else {
-        // eslint-disable-next-line no-param-reassign
-        files[file].headerFooterData = serialized;
-      }
```
Comment (ncksllvn) [inline src/site/stages/build/plugins/create-header-footer.js:68]: There was a bunch of old code here used when only a small segment of the site was building from Drupal. On those pages, the build would inject the  headerFooter data from Drupal but on the other pages (from vagov-content) we would continue to use the headerFooter data we had on GitHub. Obviously, the whole site now uses the headerFooter data from Drupal, so I just removed the conditions to flatten the logic.

File: src/site/stages/build/plugins/create-header-footer.js [hunk 3]
```diff
@@ -51,31 +18,19 @@ function createHeaderFooterData(buildOptions) {
       megaMenuData,
     };
 
-    const serialized = JSON.stringify(headerFooter, null, 4);
-
-    const drupalMenu = DRUPALS.PREFIXED_ENVIRONMENTS.has(buildOptions.buildtype)
-      ? replaceWithDrupalLinks(headerFooter, files)
-      : headerFooter;
```
Comment (ncksllvn) [inline src/site/stages/build/plugins/create-header-footer.js:58]: `PREFIXED_ENVIRONMENTS` is empty and has been for a long time so this code was always a noop.

**Review by billfienberg — APPROVED** [review]

**Comment by ncksllvn** [comment]
Planning on merging this after the daily deployment so I can see it on Staging.

**Pull request merged**